--- a/lib/libipv6calc.h
+++ b/lib/libipv6calc.h
@@ -78,7 +78,7 @@
 //  case 2: src is too long to be added to dst, add only what is available but leave space for ...
 //  case 3: dst is already exhausted, override end with ...
 //  gcc 8.5 generates a false alarm because not proper analyzing the calculations
-#if __GNUC__ == 8 && __GNUC_MINOR__ == 5
+#if __GNUC__ == 8 && __GNUC_MINOR__ == 4
 
 #define STRCAT(dst, src) \
 	_Pragma("GCC diagnostic push") \
